### ISSUE TITLE
Pyglet window does not open and crashes on start-up

### ISSUE THREAD

**Issue by Andrea-Oliveri** [pyglet/pyglet#1019]
**High-level description of the bug:**
Good morning,
I have developed an application with Pyglet. When I run it from the console as such: `python ./app.py`, most of the time it works, the window opens and responds as expected. However, 1 out of 10 times approximately, after running that command, no window opens, nothing is printed on the console, but the process ends after a couple of seconds, leaving me with this:
```
>>> python ./app.py
>>>
```
More details and the debugging I managed to do on my own follow in the "How To Reproduce" section.
I would be happy to support by providing further insights and debugging further under your guidance.
Thank you very much for your help.

**System Information:**
Here is the output of `python -m pyglet.info`:
```
Platform
------------------------------------------------------------------------------
platform:  Windows-11-10.0.22631-SP0
release:   11
machine:   AMD64

Python
------------------------------------------------------------------------------
implementation: CPython
sys.version: 3.12.0 | packaged by Anaconda, Inc. | (main, Oct  2 2023, 17:20:38) [MSC v.1916 64 bit (AMD64)]
sys.maxint: 9223372036854775807
os.getcwd(): C:\Users\andre\Desktop\Python Pac-Man

pyglet
------------------------------------------------------------------------------
pyglet.version: 2.0.10
pyglet.compat_platform: win32
pyglet.__file__: C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\__init__.py
pyglet.options['audio'] = ('xaudio2', 'directsound', 'openal', 'pulse', 'silent')
pyglet.options['debug_font'] = False
pyglet.options['debug_gl'] = True
pyglet.options['debug_gl_trace'] = False
pyglet.options['debug_gl_trace_args'] = False
pyglet.options['debug_gl_shaders'] = False
pyglet.options['debug_graphics_batch'] = False
pyglet.options['debug_lib'] = False
pyglet.options['debug_media'] = False
pyglet.options['debug_texture'] = False
pyglet.options['debug_trace'] = False
pyglet.options['debug_trace_args'] = False
pyglet.options['debug_trace_depth'] = 1
pyglet.options['debug_trace_flush'] = True
pyglet.options['debug_win32'] = False
pyglet.options['debug_input'] = False
pyglet.options['debug_x11'] = False
pyglet.options['shadow_window'] = True
pyglet.options['vsync'] = None
pyglet.options['xsync'] = True
pyglet.options['xlib_fullscreen_override_redirect'] = False
pyglet.options['search_local_libs'] = True
pyglet.options['win32_gdi_font'] = False
pyglet.options['headless'] = False
pyglet.options['headless_device'] = 0
pyglet.options['win32_disable_shaping'] = False
pyglet.options['dw_legacy_naming'] = False
pyglet.options['win32_disable_xinput'] = False
pyglet.options['com_mta'] = False
pyglet.options['osx_alt_loop'] = False

pyglet.window
------------------------------------------------------------------------------
display: <pyglet.canvas.win32.Win32Display object at 0x000002D01826E9C0>
screens[0]: Win32Screen(x=0, y=0, width=1536, height=864)
```

**How To Reproduce:**

_Structure of the application_
```
APP_FOLDER
    |--- app.py
    |--- src
    |     |--- window.py
    |     |--- /* other scripts */
    |--- assets
            |--- images
            |      |--- ...
            |--- sounds
                   |--- /* currently empty*/
```

*Content of app.py*
```
# -*- coding: utf-8 -*-

import os
import sys

# Needed for compatibility with pyinstaller.
if getattr(sys, 'frozen', False):
    os.chdir(sys._MEIPASS)

import pyglet.app
from src.window import Window

window = Window()

pyglet.app.run()
```

*Content of window.py*
```
# -*- coding: utf-8 -*-

import pyglet
# /* other imports */

class Window(pyglet.window.Window):
    def __init__(self):
        super().__init__(**{'width': 250,
                            'height': 300,
                            'fullscreen': False,
                            'resizable': True,
                            'caption': "App-Name",
                            'style': Window.WINDOW_STYLE_DEFAULT,
                            'vsync': True})
        
        self.set_minimum_size(250, 300)

        icon = pyglet.image.load("./assets/images/icon.ico")
        self.set_icon(icon)
        
        # /* other stuff to initialize*/

    def on_resize(self, width, height):
        # /* implemented */
    
    def on_key_press(self, symbol, modifiers):
        # /* implemented */

    def on_draw(self):
        # /* implemented */
```

_What I could debug:_
I have made extensive use of `print` to trace the line where the program hangs.
The culprit is line `super().__init__(**{.....})` inside of `window.py`. Indeed, the `print` function correctly printed before (inside the constructor of my `Window` class) but not after this line.

Finally, I have attempted to use Pyglet's [runtime options](https://pyglet.readthedocs.io/en/latest/programming_guide/options.html) to try understanding what could be happening, but I could not interpret what they provided me. I have added the following lines on top of my `app.py`:

```
for o in ('debug_gl', 'debug_gl_shaders', 'debug_gl_trace', 'debug_gl_trace_args',
          'debug_graphics_batch', 'debug_input', 'debug_lib', 'debug_media', 'debug_texture',
          'debug_trace', 'debug_trace_args', 'debug_win32'):
    pyglet.options[o] = True
```
Attached are two text files, one with the output for when the window runs as expected, and one for when it does not run as expected.
[Run Fail.txt](https://github.com/pyglet/pyglet/files/13731017/Run.Fail.txt)
[Run Success.txt](https://github.com/pyglet/pyglet/files/13731018/Run.Success.txt)

**Comment by pushfoo** [comment]
```
sys.version: 3.12.0 | packaged by Anaconda, Inc. | (main, Oct  2 2023, 17:20:38) [MSC v.1916 64 bit (AMD64)]
```
Can you try with a different Python version such as 3.10 or 3.11? This may be a 3.12 issue (see #961).

**Comment by caffeinepills** [comment]
Your pyglet.info is incomplete and doesn't show the OpenGL or graphics driver, which means it's probably crashing there as well.

I would check your graphics drivers and make sure they are up to date. Also I've seen numerous tickets with issues with Anaconda environments, I would make sure to try your scripts outside that environment to ensure it's not that. Also I would try Python 3.11 to make sure it's not an issue there. It could also be a combination of all of these. Let us know.

**Comment by Andrea-Oliveri** [comment]
I have tried in a freshly created conda environment with Python 3.10. Unfortunately, can confirm the issue appeared in this environment as well (tested both in my app and in the very simple script mentioned at the end of the previous comment).

Please find the output of `python -m pyglet.info`:
```
Platform
------------------------------------------------------------------------------
platform:  Windows-10-10.0.22631-SP0
release:   10
machine:   AMD64

Python
------------------------------------------------------------------------------
implementation: CPython
sys.version: 3.10.13 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:24:38) [MSC v.1916 64 bit (AMD64)]
sys.maxint: 9223372036854775807
os.getcwd(): C:\Users\andre\Desktop\Pyglet issue test

pyglet
------------------------------------------------------------------------------
pyglet.version: 2.0.10
pyglet.compat_platform: win32
pyglet.__file__: C:\Users\andre\miniconda3\envs\pyglet_issue\lib\site-packages\pyglet\__init__.py
pyglet.options['audio'] = ('xaudio2', 'directsound', 'openal', 'pulse', 'silent')
pyglet.options['debug_font'] = False
pyglet.options['debug_gl'] = True
pyglet.options['debug_gl_trace'] = False
pyglet.options['debug_gl_trace_args'] = False
pyglet.options['debug_gl_shaders'] = False
pyglet.options['debug_graphics_batch'] = False
pyglet.options['debug_lib'] = False
pyglet.options['debug_media'] = False
pyglet.options['debug_texture'] = False
pyglet.options['debug_trace'] = False
pyglet.options['debug_trace_args'] = False
pyglet.options['debug_trace_depth'] = 1
pyglet.options['debug_trace_flush'] = True
pyglet.options['debug_win32'] = False
pyglet.options['debug_input'] = False
pyglet.options['debug_x11'] = False
pyglet.options['shadow_window'] = True
pyglet.options['vsync'] = None
pyglet.options['xsync'] = True
pyglet.options['xlib_fullscreen_override_redirect'] = False
pyglet.options['search_local_libs'] = True
pyglet.options['win32_gdi_font'] = False
pyglet.options['headless'] = False
pyglet.options['headless_device'] = 0
pyglet.options['win32_disable_shaping'] = False
pyglet.options['dw_legacy_naming'] = False
pyglet.options['win32_disable_xinput'] = False
pyglet.options['com_mta'] = False
pyglet.options['osx_alt_loop'] = False

pyglet.window
------------------------------------------------------------------------------
display: <pyglet.canvas.win32.Win32Display object at 0x000001FE90F94E50>
screens[0]: Win32Screen(x=0, y=0, width=1536, height=864)
config['double_buffer'] = 1
config['stereo'] = 0
config['buffer_size'] = 32
config['aux_buffers'] = 4
config['sample_buffers'] = 0
config['samples'] = 0
config['red_size'] = 8
config['green_size'] = 8
config['blue_size'] = 8
config['alpha_size'] = 8
config['depth_size'] = 24
config['stencil_size'] = 8
config['accum_red_size'] = 0
config['accum_green_size'] = 0
config['accum_blue_size'] = 0
config['accum_alpha_size'] = 0
config['major_version'] = 3
config['minor_version'] = 3
config['forward_compatible'] = None
config['opengl_api'] = 'gl'
config['debug'] = None
context: Win32ARBContext(id=2192833788848, share=Win32Context(id=2192865582320, share=None))

window.context._info
------------------------------------------------------------------------------
gl_info.get_version(): (3, 3)
gl_info.get_vendor(): ATI Technologies Inc.
gl_info.get_renderer(): AMD Radeon(TM) Graphics

pyglet.gl.glx_info
------------------------------------------------------------------------------
GLX not available.

pyglet.media
------------------------------------------------------------------------------
audio driver: <pyglet.media.drivers.xaudio2.adaptation.XAudio2Driver object at 0x000001FE97EBCAC0>

pyglet.media.ffmpeg
------------------------------------------------------------------------------
FFmpeg not available.

pyglet.media.drivers.openal
------------------------------------------------------------------------------
OpenAL not available.

pyglet.input.wintab
------------------------------------------------------------------------------
WinTab not available.
```

At this time I am not in a situation which allows me to easily test the app outside of a conda environment. Are there any actions we can take towards understanding if that may indeed be the cause of the issue? Perhaps if I packaged the interpreter with Pyinstaller, would that be worth trying? Or perhaps some debug options?

Please also find the exact conda environment I am operating in, for future reference:
```
# Name                    Version                   Build  Channel
bzip2                     1.0.8                he774522_0
ca-certificates           2023.12.12           haa95532_0
libffi                    3.4.4                hd77b12b_0
openssl                   3.0.12               h2bbff1b_0
pip                       23.3.1          py310haa95532_0
pyglet                    2.0.10                   pypi_0    pypi
python                    3.10.13              he1021f5_0
setuptools                68.2.2          py310haa95532_0
sqlite                    3.41.2               h2bbff1b_0
tk                        8.6.12               h2bbff1b_0
tzdata                    2023c                h04d1e81_0
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
wheel                     0.41.2          py310haa95532_0
xz                        5.4.5                h8cc25b3_0
zlib                      1.2.13               h8cc25b3_0
```

**Comment by Andrea-Oliveri** [comment]
I have made another observation in my attempts debugging, which I hope will be useful to further understand how to dig deeper. I have read about the `shadow_window`, pretty much by accident.

I have vaguely understood what its purpose is, but not how to use it (and found no examples online, which I would therefore suggest to add to the documentation). I have however grasped that it's only useful if I had multiple windows (a.k.a. OpenGL contexts) in my app (which I don't), and that some drivers don't appreciate it.

I have therefore tried to disable it using the global runtime options: `pyglet.options['shadow_window'] = False`.

In both of my test apps, the following behaviours were observed:

1) The error above does not happen anymore (can't claim this with 100% certainty, but I can say that I have tried many times re-running the apps and the issue did not happen a single time)
2) Another issue that I thought was unrelated disappeared as well. That one is described below.

**Other mysterious issue:**
In a similar fashion to the original issue in the first comment, this one also does not happen all the time. Sometimes, after closing the window of my application (the only window I ever explicitly instantiated), despite the window disappearing from the screen and the taskbar, the console remains unusable for 3-4 seconds, hinting that the process is not over yet. Some other times, it just ends right away.

*Please let me know if these new findings are useful and how you would like to advance in debugging this issue.
Thank you again for your support.*

**Comment by caffeinepills** [comment]
Unfortunately I still have yet to reproduce this.

I have cleaned up the traces to give information on which functions are being called in the latest versions, but for now I will mark this as unreproducible.

**Comment by eschan145** [comment]
I've had bugs with Python stopping suddenly for no apparent reason and without any error message whatsoever, as if `sys.exit()` was called. What you are experiencing is almost certainly some sort of memory error such as a segmentation fault; it may be helpful if you could run your script with `py -X faulthandler -m app.py` and see what the faulthandler finds.

**Comment by Andrea-Oliveri** [comment]
Good afternoon,
thanks for still looking into this 😄 .
I have run the app with `faulthandler` and here is the output:

```
Windows fatal exception: code 0xc0000374

Current thread 0x00000f78 (most recent call first):
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 714 in f
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 250 in _create
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\__init__.py", line 578 in __init__
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 104 in __init__
  File "C:\Users\andre\Desktop\Pac-Man\src\window.py", line 23 in __init__
  File "C:\Users\andre\Desktop\Pac-Man\pacman.py", line 36 in <module>

```

**Comment by caffeinepills** [comment]
Just curious, does adding a caption to the window (or removing the caption) any impact on it?

**Comment by pushfoo** [comment]
**TL;DR:** Check which version of the win32 call (ASCII vs wide) should take, the ctypes binding's match for that, and whether non-anaconda changes things?

The last one's weird, I know, but let's cover our bases with elaboration on what caffeinepills said?

> [!NOTE]
> I'm not on win32 atm and it's [been a little bit](https://github.com/pushfoo/pyc2e) since I have been, so I might be wrong.

As to "two or more" function versions in the Microsoft doc, Windows has a bad habit of having a "wide" and non-wide ASCII (`FuncNameA`) version of many functions. Which one are we binding below, and is it getting the correct types? Is it passing out the correct types?

![Image](https://github.com/user-attachments/assets/f1cb2ce2-ceeb-4c32-a4bd-7d19c8518f01)

The last one's a long shot, but if we run out of options, let's eliminate it too.

**Comment by Andrea-Oliveri** [comment]
Good evening,
### Section 1: new crash observed
while running the tests you requested I noticed the following: sometimes, after the window opened correctly, when I close it the program takes a long time to actually end (I can see because I don't regain control of the terminal).
With `faulthandler` enabled, when this slowdown appears, the following traceback is also printed:
```
Windows fatal exception: code 0xc0000374

Current thread 0x00004690 (most recent call first):
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\gl\win32.py", line 211 in detach
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\gl\base.py", line 297 in destroy
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\__init__.py", line 659 in close
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 299 in close
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\__init__.py", line 799 in on_close
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\event.py", line 387 in dispatch_event
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\__init__.py", line 668 in dispatch_event
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 1037 in _event_close
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 708 in f
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 714 in f
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 714 in f
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\app\win32.py", line 89 in step
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\app\base.py", line 152 in run
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\app\__init__.py", line 76 in run
  File "C:\Users\andre\Desktop\Pac-Man\pacman.py", line 39 in <module>
```
It would therefore appear that the issue is not isolated to window creation, but also to window destruction.

### Section 2: drivers and display settings
On the topic of old drivers, I don't think it's likely because I keep my whole PC up to date and it's now been >1 year since I first reported this. Buggy, may be 😅 . I also unfortunately did not understand how I could test the comment on _different scaling, monitor, or DPI configuration_.

### Section 3: removing the caption
I tried removing the caption (and also the icon, for good measure). Unfortunately this bug is non-deterministic so I can't say for certain, but I feel the crash on startup appears less often. I nonetheless had it at least once, here is the traceback (notice it slightly differs from the one I posted on 10th May 2025):
```
Windows fatal exception: code 0xc0000374

Current thread 0x0000336c (most recent call first):
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\gl\lib_wgl.py", line 66 in __call__
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\gl\win32.py", line 93 in _get_arb_pixel_format_matching_configs
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\gl\win32.py", line 22 in match
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\canvas\win32.py", line 36 in get_matching_configs
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\canvas\base.py", line 162 in get_best_config
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\__init__.py", line 523 in __init__
  File "C:\Users\andre\miniconda3\envs\pacman\Lib\site-packages\pyglet\window\win32\__init__.py", line 104 in __init__
  File "C:\Users\andre\Desktop\Pac-Man\src\window.py", line 23 in __init__
  File "C:\Users\andre\Desktop\Pac-Man\pacman.py", line 37 in <module>
```
The crash on closing, however, appeared definitely just as often, with the same traceback as shown at the start of this message.

### Section 4: new pyglet version
I have created a new conda environment as such:
```
conda create -n pacman_debug python=3.11.4
conda activate pacman_debug
pip install pyglet==2.1.6
``` 
As you can see, I kept the Python version constant and upped pyglet to the latest version available in `pip`.

The crashes definitely still happen. At start-up:
```
Windows fatal exception: code 0xc0000374

Thread 0x00004058 (most recent call first):
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\threading.py", line 320 in wait
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\threading.py", line 622 in wait
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\media\player_worker_thread.py", line 55 in run
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\threading.py", line 1038 in _bootstrap_inner
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\threading.py", line 995 in _bootstrap

Current thread 0x0000194c (most recent call first):
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\win32\__init__.py", line 843 in f
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\win32\__init__.py", line 307 in _create
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\__init__.py", line 577 in __init__
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\win32\__init__.py", line 153 in __init__
  File "C:\Users\andre\Desktop\Pac-Man\src\window.py", line 23 in __init__
  File "C:\Users\andre\Desktop\Pac-Man\pacman.py", line 37 in <module>
```

and at closure, I actually got two different ones on different runs:
```
Windows fatal exception: code 0xc0000374

Thread 0x00002b80 (most recent call first):
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\libs\win32\com.py", line 208 in pinterface_method_forward
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\media\drivers\xaudio2\interface.py", line 289 in _delete_driver
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\media\drivers\xaudio2\adaptation.py", line 49 in delete
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\media\drivers\__init__.py", line 84 in _delete_audio_driver
```
and:
```
Windows fatal exception: code 0xc0000374

Thread 0x000046b4 (most recent call first):
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\threading.py", line 320 in wait
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\threading.py", line 622 in wait
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\media\player_worker_thread.py", line 55 in run
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\threading.py", line 1038 in _bootstrap_inner
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\threading.py", line 995 in _bootstrap

Current thread 0x0000556c (most recent call first):
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\gl\win32.py", line 201 in detach
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\gl\base.py", line 351 in destroy
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\__init__.py", line 663 in close
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\win32\__init__.py", line 356 in close
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\__init__.py", line 834 in on_close
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\event.py", line 368 in dispatch_event
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\__init__.py", line 672 in dispatch_event
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\win32\__init__.py", line 1167 in _event_close
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\win32\__init__.py", line 837 in f
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\win32\__init__.py", line 843 in f
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\window\win32\__init__.py", line 843 in f
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\app\win32.py", line 89 in step
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\app\base.py", line 165 in run
  File "C:\Users\andre\miniconda3\envs\pacman_debug\Lib\site-packages\pyglet\app\__init__.py", line 79 in run
  File "C:\Users\andre\Desktop\Pac-Man\pacman.py", line 39 in <module>
```
Attached are the runs:
[Success.txt](https://github.com/user-attachments/files/20232450/Success.txt)
[Startup Failure.txt](https://github.com/user-attachments/files/20232453/Startup.Failure.txt)
[Closing Failure.txt](https://github.com/user-attachments/files/20232455/Closing.Failure.txt)
[debug_win32.log](https://github.com/user-attachments/files/20232468/debug_win32.log) (I really hope this last log has all the runs, I noticed it too late. Let me know if I need to remake one for each of success, startup failure, closing failure)

### Section 5: Python outside conda
I installed Python 3.11.4 from the official binaries and pyglet 2.1.6
Exactly the same crashes occur with exactly the same tracebacks.

### Section 6: Thank you
I would like to reiterate my thanks for maintaining this great library. I am genuinely happy to help as best I can. Nonetheless, a reminder for whomever may be reading this in the future (and perhaps may also be worth putting in the documentation?), I had originally solved my problem by running
```
pyglet.options['shadow_window'] = False
```
why that solves it, I have no idea.

Let me know if you'd like me to try some more stuff out to pinpoint this problem 😄

**Comment by caffeinepills** [comment]
If disabling the shadow window fixes it, I will close this.

In 3.0 (development) removes the shadow window, so theoretically this should resolve the issue. If the problem still persists, we can try another approach.
